data/basics/teal.go

```diff
@@ -123,80 +123,6 @@ func (sd StateDelta) Valid(proto *config.ConsensusParams) error {
 	return nil
 }
 
-// LogItem is contains logs for an application
-// ID is the offset into Txn.ForeignApps
-type LogItem struct {
-	_struct struct{} `codec:",omitempty,omitemptyarray"`
-
-	ID      uint64 `codec:"i"`
-	Message string `codec:"m"`
-}
-
-// Equal checks whether two LogItems are equal.
-func (l LogItem) Equal(o LogItem) bool {
-
-	return l.ID == o.ID && l.Message == o.Message
-
-}
-
-// EvalDelta stores StateDeltas for an application's global key/value store, as
-// well as StateDeltas for some number of accounts holding local state for that
-// application
-type EvalDelta struct {
```

Comment:
It's pretty transaction specific.  At any rate, I moved it to break a circular reference (it contains a transactions.SignedTransactionWithAD now) rather than out of a strong belief that it "belongs" here.